k-distribution/include/kframework/builtin/domains.md

```diff
@@ -1055,7 +1059,13 @@ module INT-SYMBOLIC [symbolic]
   rule X modInt N => X requires 0 <=Int X andBool X <Int N [simplification]
   rule X   %Int N => X requires 0 <=Int X andBool X <Int N [simplification]
 
-  rule _ ^Int N => 0 requires N <Int 0 [simplification]
+  rule  1 ^Int _ =>  1                           [simplification]
+  rule -1 ^Int N =>  1 requires N %Int 2 ==Int 0 [simplification]
+  rule -1 ^Int N => -1 requires N %Int 2 ==Int 1 [simplification]
+  rule  0 ^Int N =>  0 requires N >Int 0         [simplification]
+  rule  X ^Int N =>  0 requires absInt(X) >Int 0
+                        andBool N <Int 0         [simplification]
+  rule  X ^Int 0 =>  1 requires X =/=Int 0       [simplification]
```

Comment:
@ehildenb @daejunpark I've implemented the special cases for integer exponentiation per the [document here](https://github.com/runtimeverification/security/blob/61e19e0830960dc048f7dd8e4f42b06be186e792/knowledge-base/exponentiation.md).

The SMT encoding will only send `_^Int_` to Z3 if both `X` and `N` are strictly positive. In the cases where that might not be the case, these simplification rules will apply if we can define the result.

Cases where the result of the exponentiation would be ambiguous (`0 ^ 0`) or infinite (`0 ^ -1`) are left undefined; if users need to define these cases they can supply their own axioms.